repositories/QuestionConfirmationRepository.test.js

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable camelcase */
+
+const db = require("../db");
+
+const buildTestQuestionnaire = require("../tests/utils/buildTestQuestionnaire");
+
+const QuestionConfirmationRepository = require("./QuestionConfirmationRepository");
+
+describe("QuestionConfirmationRepository", () => {
+  let page;
+  beforeAll(async () => {
+    await db.migrate.latest();
+    const questionnaire = await buildTestQuestionnaire({
+      sections: [
+        {
+          pages: [
+            {
+              answers: []
+            }
+          ]
+        }
+      ]
+    });
+    page = questionnaire.sections[0].pages[0];
+  });
+  afterAll(() => db.destroy());
+  afterEach(() => db("QuestionConfirmations").delete());
+
+  describe("create", () => {
+    it("should create and return the confirmation", async () => {
+      const confirmation = await QuestionConfirmationRepository.create({
+        pageId: page.id
+      });
+      expect(confirmation).toMatchObject({
+        id: expect.any(Number),
+        title: null,
+        pageId: page.id,
+        positiveLabel: null,
+        positiveDescription: null,
+        negativeLabel: null,
+        negativeDescription: null,
+        isDeleted: false,
+        created_at: expect.any(Date),
+        updated_at: expect.any(Date)
+      });
+    });
+
+    it("should not allow you to create a confirmation for a question if one already exists", async () => {
+      await QuestionConfirmationRepository.create({
+        pageId: page.id
+      });
+      let thrown = false;
+      try {
+        await QuestionConfirmationRepository.create({
+          pageId: page.id
+        });
+      } catch (e) {
+        thrown = true;
+      }
+      expect(thrown).toBeTruthy();
```

Comment:
Would it be slightly more reliable to assert on the error message rather than just an error has been thrown?

Reply:
It now asserts on the message as well.